platform/pulp/app/models/storage.py

```diff
@@ -10,10 +11,98 @@
 from django.utils.deconstruct import deconstructible
 
 
+class FileContent(File):
+    """
+    An in-memory file.
+    Designed to be used by FileField when content is known.
+
+    Attributes:
+        content (str): The actual content.
+
+    Examples:
+        >>>
+        >>> from django.db import models
+        >>>
+        >>> class Person(models.Model):
+        >>>     ssl_certificate = models.FileField()
+        >>>
+        >>> person = Person()
+        >>> certificate = 'PEM ENCODED CERTIFICATE HERE'
+        >>> person.ssl_certificate = FileContent(certificate)
+        >>>
+    """
+
+    def __init__(self, content):
+        """
+        Args:
+            content (str): The file content.
+
+        Notes:
+            The name is set to '-' to ensure that model.FileField
+            will work properly.
```

Comment:
More info would be helpful. Something like: For a regular Django `File` object, `name` is the path to the file-like object. Our `content` is already in memory, so `name` isn't necessary to open the file.